### PR TITLE
ft-buyer-signup #3

### PR DESCRIPTION
**Pull request by patsicko**
# What this PR do
# Create a buyer signup endpoint
# Description of the task
- Create user model
- Create buyer controller
- Create buyer api endpoint
- Test buyer api with jest
- Document buyer api with swagger

## How should this be manually tested

- clone the repo
- use npm install to install all packages
- Add all required environment variables following the .env.example provided
- use npx sequelize db:migrate to migrate migrations
- use npm run dev to start the server
use the appropriate router in postman or thunderclient 
navigate to  the http://localhost:PORT  by replacing 'PORT' the port that your the app is listening to, and test it using the documented API. By sending a request to create a buyer, the response will be displayed containging a newly created buyer and welcome message will be sent to the provided email
- Run npm test to test the app

## Trello story id
#003

### CODE REVIEW

**Review by sostene12 — COMMENTED** [review]
you can add more details on this pull, like profile model and work hand in hand with task of aimable

**Review by sostene12 — COMMENTED** [review]
- please solve the conflicts

**Review by sostene12 — APPROVED** [review]

**Review by wakaflorien — CHANGES_REQUESTED** [review]
- **TEST_DB_HOST, TEST_DB_USERNAME, TEST_DB_PASSWORD** are not needed.
- existing on conflict error codes should be 409
- no need to push console logs in your codes once they served your purpose
- profile model still unnormalized e.g how is **businessNumber** attribute relevant to **buyer**. same goes to **businessAddress**, **taxIdNumber, typeOfProducts**
- the PR description doesn't well guide to how this can be manually tested

**Review by chrill-christella — APPROVED** [review]

File: src/controllers/buyerController.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    const buyer = await new user({
+      firstName,
+      lastName,
+      email,
+      password:hashedPassword
+    });
+console.log(buyer);
+    buyer.save();
+
+    const token = jwt.sign({ id: buyer.id }, process.env.JWT_SECRET, {
```
Comment (wakaflorien) [inline src/controllers/buyerController.js:42]: - when the JWT_SECRET is no found the account is still created.

File: test/user.test.js [hunk 2]
```diff
@@ -8,14 +8,47 @@ describe("testing all routes", () => {
...
+    const response3=await request(app).post("/buyerSignup").send({
+      
+      firstName: "umuntu",
+      lastName: "Person",
+      email: "umuntu10@gmail.com",
+      password: "1234567@password"
+    
+    });
+
+ console.log(response.body);
```
Comment (wakaflorien) [inline test/user.test.js]: - no need of leaving this console logs @patsicko

**Review by wakaflorien — CHANGES_REQUESTED** [review]

**Review by wakaflorien — APPROVED** [review]

**Pull request merged**